torch/profiler/profiler.py

```diff
@@ -16,16 +16,19 @@ class ProfilerAction(Enum):
     RECORD_AND_SAVE = 3
 
 
-def schedule(*, wait: int, warmup: int, active: int) -> Callable:
+def schedule(*, wait: int, warmup: int, active: int, repeat: int = 0) -> Callable:
```

Comment:
How do I get a trace collection on step 100 followed by 1100 and 2100?
I.e. I want three traces collected, the first after some workload warmup period and then a couple of more after some fixed intervals.

Reply:
You may need to write a schedule by yourself. It return ProfilerAction.WARMUP at beginning and return ProfilerAction.RECORD at step 100, 1100, 2100, and ProfilerAction.NONE at other.

Reply:
I agree that a one-shot is much more useful in practice as the default. And the above use case can almost be supported with wait = 1000, repeat = 3.

Reply:
Right, I just question whether this API is very useful, other than as a one-shot. On the one hand you typically want to collect a trace as soon as the workload is stable and then perhaps a few times during a longer training run.
When would you use a non-zero repeat argument in practice with this API?

Reply:
I agree. A one-shot is much more useful and this api will get the one-shot result when repeat=1. If there's no this 'repeat' argument, this profiling cycles will continue until the training loop finish. Do you mean we should set repeat=1 as default? Then this api will do the one-shot profiling by default.

Reply:
Right, that's what I meant yes, repeat = 1 will be more useful.

Reply:
Replace repeat of 0 with 1 in my previous comments :)

Reply:
This pr has merged..We can do it in another pr...

Reply:
Sure, we can discuss and refine.